editor/src/components/editor/persistence/persistence.ts

```diff
@@ -133,7 +134,8 @@ export class PersistenceMachine {
 
     this.interpreter.start()
 
-    window.addEventListener('beforeunload', (e) => {
+    window.addEventListener('beforeunload', async (e) => {
+      await releaseControl()
```

Comment:
I don't think this should be using an `await` because the browser would just let the page close anyway, meaning the code below might not execute if this takes too long. Also, if the browser does close the page too quickly for this to execute, will control still be released server side?

Reply:
`visibilitychange` is generally safer than `beforeunload`: https://developer.mozilla.org/en-US/docs/Web/API/Window/beforeunload_event#usage_notes

Reply:
@gbalint Does that mean this save hook should be using that instead?

Reply:
@seanparsons yes, I think so

Reply:
Fixed in fcc60802d.